FirebaseCore/Sources/FIRApp.m

```diff
@@ -836,7 +836,8 @@ + (void)registerSwiftComponents {
   SEL componentsToRegisterSEL = @selector(componentsToRegister);
   // Dictionary of class names that conform to `FIRLibrary` and their user agents. These should only
   // be SDKs that are written in Swift but still visible to ObjC.
-  NSDictionary<NSString *, NSString *> *swiftLibs = @{@"FIRFunctionsComponent" : @"fire-fun"};
+  NSDictionary<NSString *, NSString *> *swiftLibs =
+      @{@"FIRFunctionsComponent" : @"fire-fun", @"FIRCombineComponent" : @"fire-comb"};
```

Comment:
Really for the Swift extensions we don't need the components behaviour, only the user agent. I think to reduce the amount of code the Swift SDKs have to write we can do another loop here with non-components code and use the `[FIRApp registerLibrary:withVersion:]` call instead. That way the Swift code only needs to be:

```
@objc(FIRCombineSDK) private class __CombineSDK: NSObject {}
```

Or you may need to add an init, but it can be empty or a fatalerror. The version can be pulled from the wider Firebase version.

Reply:
I added another loop to iterate the non-component libraries and check if the Combine SDK "component" is present.